src/protocols/MeterS0.cpp

```diff
@@ -322,7 +322,7 @@ ssize_t MeterS0::read(std::vector<Reading> &rds, size_t n) {
 
 	if (_send_zero || t_imp > 0) {
 		if (!_first_impulse) {
-			double value = 3600000 / ((t2-t1) * (_resolution * t_imp));
```

Comment:
My logic was that the previous term was ok for t_imp==1 so I just wanted to multiply with t_imp the value.